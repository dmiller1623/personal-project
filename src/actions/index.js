export const searchRelated = (searchedRelated) => ({
  type: 'SEARCH_RELATED',
  searchedRelated
}) 