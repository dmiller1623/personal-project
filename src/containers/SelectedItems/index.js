// import React, { Component } from 'react'
// import { connect } from 'http2';
// import { mapStateToProps, mapDispatchToProps } from 'react-redux';
// import SelectedItemsContainer from '../../components/SelectedItemsContainer'

// class SelectedItems extends Component{
//   constructor() {
//     super();
//   }

//   render() {
//     const { additionalResources } = this.props
//     return(
//       <div>
//         <SelectedItemsContainer additionalResources={additionalResources}/>
//       </div>
//     )
//   }
// }

// export const mapStateToProps = (state) => ({
//   additionalResources: state.additionalResources
// })

// export default connect(mapStateToProps, mapDispatchToProps)(SelectedItems)