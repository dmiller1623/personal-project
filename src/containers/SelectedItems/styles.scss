.selected-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-wrap: wrap;
  display: flex;
  width: 100%;
  height: 100vh;
  overflow: scroll;
}