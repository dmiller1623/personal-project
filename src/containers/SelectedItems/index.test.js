import React from 'react';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps, SelectedItems } from '../SelectedItems';
import { removeResource } from '../../actions';

describe('SelectedItems', () => {
  let wrapper;
  describe('SelectedItems Container', () => {
    it('should match the snapshot', () => {
      wrapper = shallow(<SelectedItems store={{}}/>);
      expect(wrapper).toMatchSnapshot();
      
    });

    it('should call removeResources when deleteResources is called', () => {
      const resource = { name: 'jim morrison'};
      const additionalResources = ['jim morrison'];
      const removeResourceMock = jest.fn();
      wrapper = shallow(<SelectedItems additionalResources={additionalResources} removeResource={removeResourceMock}/>);

      wrapper.instance().deleteResources(resource);
      expect(removeResourceMock).toHaveBeenCalled();
    });

    it('should not call removeResources if the resource down not exsist', () => {
      const resource = { name: 'jim morrison'};
      const additionalResources = [];
      const removeResourceMock = jest.fn();
      wrapper = shallow(<SelectedItems additionalResources={additionalResources} removeResource={removeResourceMock}/>);

      wrapper.instance().deleteResources(resource);
      expect(removeResourceMock).not.toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an array of additional resources', () => {
      const store = {
        searchedResources: [],
        additionalResources: []
      };
  
      const expected = {
        additionalResources: []
      };
  
      const mappedProps = mapStateToProps(store);
      expect(mappedProps).toEqual(expected);

    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function MTDP', () => {
      const mockedDispatch = jest.fn();
      const actionToDispatch = removeResource();
      const mappedProps = mapDispatchToProps(mockedDispatch);

      mappedProps.removeResource();
      expect(mockedDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

  });

});