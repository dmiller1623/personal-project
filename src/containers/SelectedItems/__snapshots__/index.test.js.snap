// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedItems SelectedItems Container should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectedItems
    store={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <SelectedItemsContainer
        additionalResources={undefined}
        deleteResources={[Function]}
/>,
      "className": "selected-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "additionalResources": undefined,
        "deleteResources": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <SelectedItemsContainer
          additionalResources={undefined}
          deleteResources={[Function]}
/>,
        "className": "selected-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "additionalResources": undefined,
          "deleteResources": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
